File: CS214-Programming Language Concepts/projects/04/c++/fact.cpp
/* fact.cpp displays the factorial of a positive number n.
 * Input: unsigned n
 * PRE: positive integer
 * Output: factorial of the number
 * Begun by Professor Adams? for CS 214 at Calvin COllege
 * Edited by Joshua Maguire
 ********************************************************/



 #include <iostream>
   using namespace std;
/* factorial computes the factorial of a positive number n.
 * Input: unsigned n
 * PRE: positive integer
 * Output: factorial of the number
 *****************************************************/
   double factorial(unsigned n) {
      double answer = 1.0;

      for (int i = 2; i <= n; i++) {
         answer *= i;
      }

      return answer;
   }


   int main() {
      cout << "\nTo compute n!, enter n: ";
      unsigned n;
      cin >> n;

      cout << n << "! = " << factorial(n) << endl;
   }

File: CS214-Programming Language Concepts/projects/04/ruby/fact.rb
# fact.cpp displays the factorial of a positive number n.
# Input: unsigned n
# PRE: positive integer
# Output: factorial of the number
# Begun by Professor Adams? for CS 214 at Calvin COllege
# Edited by Joshua Maguire
#*******************************************************/


# computeFact computes the factorial of a positive number n.
# Input: unsigned n
# PRE: positive integer
# Output: factorial of the number
def computeFact(n) #declare..
	result=1 #set result to default 1
	for i in 2..n #for loop
		result=result*i
	end 
	puts result #print result
end


#main
if __FILE__ == $0
	print "\nTo compute n!, enter n: "
	n = gets.chomp.to_i
	print n
	print "! = "
	computeFact(n)
end
